phdi/fhir/tabulation/tables.py

```diff
@@ -356,44 +356,58 @@ def _generate_search_urls(schema: dict) -> dict:
     return url_dict
 
 
-def drop_invalid(data: List[list], schema_columns: dict) -> List[list]:
+def drop_invalid(data: Dict, schema: Dict) -> List[list]:
     """
-    Removes resources from FHIR response if the resource contains an invalid value for
-    fields where the field's value is invalid, as specified in the invalid_values field
-    in a user-defined schema. Users may provide invalid values as a list, including
-    empty string values ("") and None/null values (null).
-
-    :param data: List of resources returned from FHIR API.
-    :param schema_columns: Dictionary of columns to include in tabulation that specifies
-      which columns should include_invalid.
-    :param return: List of resources with invalid resources.
+    Removes resources from tabulated data if the resource contains an invalid value, as
+    specified in the invalid_values field in a user-defined schema. Users may provide
+    invalid values as a list, including empty string values ("") and
+    None/null values (null).
+
+    :param data:  A dictionary mapping table names to lists of lists. The first list in
+        the data value is a list of headers serving as the columns, and all subsequent
+        lists are rows in the table.
+    :param schema: A schema of columns and values to apply to the
+      tabulated data, including invalid_values if applicable.
+    :param return: A dictionary mapping table names to lists of lists, without resources
+        that contained invalid values. The first list in the data value is a list of
+        headers serving as the columns, and all subsequent lists are rows in the table.
     """
-    # Identify columns to drop invalid values
-    invalid_columns_to_drop = [
-        column
-        for column in schema_columns.keys()
-        if schema_columns[column].get("drop_invalid", False)
-    ]
+    # Identify columns to drop invalid values for each table in schema
+    invalid_columns_to_drop = {}
+    for table in schema.get("tables"):
+        invalid_columns_to_drop[table] = [
+            column
+            for column in schema.get("tables").get(table).get("columns")
+            if (
+                schema.get("tables")
+                .get(table)
+                .get("columns")
+                .get(column)
+                .get("invalid_values")
+            )
+        ]
```

Comment:
`get()` won't throw an exception if nothing is found at that index, but it will return None. I'd suggest using a direct index for anything that you expect will always succeed, and give a default if you are handling a case that you later expect to be valued.
```suggestion
    for table in schema.get("tables",{}):
        invalid_columns_to_drop[table] = [
            column
            for column in schema["tables"][table].get("columns", {})
            if (
                schema["tables"][table]["columns"][column].get("invalid_values")
            )
        ]
```

This should
* avoid a TypeError: NoneType object is not iterable error if "tables" or "columns" are empty
* Use a slightly more readable syntax when we (developers) are confident a dict index exists
* throw a KeyError if table or column doesn't exist (which they always should since you are iterating over them)
* .get("invalid_values") will evaluate to None if it isn't values which works fine when checking for presence of that configuration